fastplotlib/graphics/_base.py

```diff
@@ -389,9 +389,11 @@ def _fpl_prepare_del(self):
         self.world_object._event_handlers.clear()
 
     def __del__(self):
-        with suppress(KeyError):
-            # remove world object if created
-            del WORLD_OBJECTS[hex(id(self))]
+        wo = WORLD_OBJECTS.pop(hex(id(self)), None)
+        # remove world object if created
+        # world object does not exist if an exception was raised during __init__ which is why this check exists
+        if wo is not None:
+            del wo
```

Comment:
Why not simply call the pop without storing the result?

Reply:
ah right that will still gc since it's no longer in locals right, an explicit `del` is not required?

Reply:
Exactly. And even if you would store the result in `wo`, it would still be gc'd once this method is done. But since you don't use `wo` it makes more sense to just not store it.

Reply:
thanks!